src/FieldWarning/Assets/Units/Component/Movement/PathfinderData.cs

```diff
@@ -49,12 +53,14 @@ public PathfinderData(TerrainMap map)
             string sceneName = SceneManager.GetActiveScene().name;
             string scenePathWithFilename = SceneManager.GetActiveScene().path;
             string sceneDirectory = Path.GetDirectoryName(scenePathWithFilename);
-            string graphFile = Path.Combine(sceneDirectory, sceneName + GRAPH_FILE_SUFFIX);
+            _graphFile = Path.Combine(sceneDirectory, sceneName + map.TerrainId + GRAPH_FILE_SUFFIX);
```

Comment:
There is a .gitignore file in Scene/ iirc, you should add the new generated files to it. This should do the job:
*_map_terrain_cache.dat
 *_map_terrain_cache.dat.meta
 *_pathfinder_graph.dat
 *_pathfinder_graph.dat.meta